testing/tidyverse_functions.R

```diff
@@ -0,0 +1,230 @@
+                                       collapse = ", "), 
+                        ")", 
+                        sep = " ")
+  
+  dataset %>% 
+    dplyr::slice_(eval)
+}
+
+# Change these varaibles for the nse function
+dataset <- stats20x.df 
+row_num <- c(1,2,3,4,6,7,8,9)
+
+datasetFilterRow.test <- datasetFilterRow(dataset, row_num)
+###---------------------------------------------------------
+
+
+# DATASET -> FILTER DATASET -> RANDOMLY
+###---------------------------------------------------------
+
+# example : take a 5 random samples each of size 3 without replacement
+filtered.RANDOM <- stats20x.df %>%
+  dplyr::sample_n(3 * 5, replace = FALSE) %>%
+    dplyr::mutate(Sample.Number = rep(1:5, each = 3))
+###
+
+
+datasetFilterRandom <- function(dataset, sample_size, n){
+  dataset %>%
+    # dplyr::mutate(row.num = 1:nrow(dataset)) %>% # uncomment this to show this is sampling without replacement
+    dplyr::sample_n(sample_size * n, replace = FALSE) %>%
+      dplyr::mutate(Sample.Number = rep(1:n, each=3))
+}
+
+# Change these varaibles for the nse function
+dataset <- stats20x.df 
+sample_size <- 3
+n <- 5
+
+datasetFilterRandom.test <- datasetFilterRandom(dataset, sample_size, n)
+###---------------------------------------------------------
+
+
+###---------------------------------------------------------
+# DATASET -> SORT DATA BY VARIABLES
+
+# example : sort data in the following order - "EXAM" (descending), "YEARSSIN" (ascending) and "DEGREE" (descending); NB. categorical variables will be sorted based on their factor levels
+sorted.3VARS <- stats20x.df %>% 
+  dplyr::arrange(desc(EXAM), YEARSSIN, desc(DEGREE))
+### 
+
+ 
+# nse function
+datasetSortVars <- function(dataset, vars, increasing = c(TRUE, TRUE, TRUE, TRUE)){
+  eval <- ifelse(increasing, 
+                 vars, 
+                 stringr::str_c("desc("
+                                , vars, 
+                                ")"))
+  dataset %>% 
+    dplyr::arrange_(.dots = as.list(eval))
+}
+
+# Change these varaibles for the nse function
+dataset <- stats20x.df 
+vars <- c("EXAM", "YEARSSIN", "DEGREE")
+increasing <- c(FALSE, TRUE, FALSE)
+
+datasetSortVars.test <- datasetSortVars(dataset, vars, increasing)
+###---------------------------------------------------------
+
+
+###---------------------------------------------------------
+# DATASET -> AGREEGATE DATA
+
+# example : get the mean, sum and sd for "ASSIGN" and median, iqr and count for "TEST" of the all unique combinations of "PASS" and "GENDER" 
+aggregated.2CATS.2NUM <- stats20x.df %>% 
+  dplyr::group_by(PASS, GENDER) %>%
+    dplyr::summarize(ASSIGN.mean = mean(ASSIGN, na.rm = TRUE), 
+                    ASSIGN.sum = sum(ASSIGN, na.rm = TRUE),
+                    ASSIGN.sd = sd(ASSIGN, na.rm = TRUE),
+                    TEST.median = median(TEST, na.rm = TRUE),
+                    TEST.count = n(),
+                    TEST.iqr = IQR(TEST, na.rm = TRUE))
+###
```

Comment:
I don't think there will be a more simple way of doing this in terms of the generated code, although we may be able to come up with something further down the track.